src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -92,9 +94,212 @@ def process_declarations(code):
+    assert ("Expecting BinaryOperation but found 'Literal'"
+            in str(excinfo.value))
+
+    operator = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND, one, one)
+    my_range = Range.create(operator, one)
+    array_reference = Array.create(symbol, [my_range])
+    with pytest.raises(InternalError) as excinfo:
+        check_bound(array_reference, 1, 0, BinaryOperation.Operator.UBOUND)
+    assert ("Expecting operator to be 'Operator.UBOUND', but found "
+            "'Operator.LBOUND'" in str(excinfo.value))
+
+    with pytest.raises(InternalError) as excinfo:
+        check_bound(array_reference, 1, 0, BinaryOperation.Operator.LBOUND)
+    assert "Expecting Reference but found 'Literal'" in str(excinfo.value)
+
+    operator = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND,
+        Reference(DataSymbol("x", DataType.INTEGER)), one)
+    my_range = Range.create(operator, one)
+    array_reference = Array.create(symbol, [my_range])
+    with pytest.raises(InternalError) as excinfo:
+        check_bound(array_reference, 1, 0, BinaryOperation.Operator.LBOUND)
+    assert ("Expecting Reference symbol 'x' to be the same as array "
+            "symbol 'a'" in str(excinfo.value))
+
+    operator = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND,
+        Reference(symbol), Node())
+    my_range = Range.create(operator, one)
+    array_reference = Array.create(symbol, [my_range])
+    with pytest.raises(InternalError) as excinfo:
+        check_bound(array_reference, 1, 0, BinaryOperation.Operator.LBOUND)
+    assert "Expecting Literal but found 'Node'" in str(excinfo.value)
+
+    operator = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND,
+        Reference(symbol), Literal("1.0", DataType.REAL))
+    my_range = Range.create(operator, one)
+    array_reference = Array.create(symbol, [my_range])
+    with pytest.raises(InternalError) as excinfo:
+        check_bound(array_reference, 1, 0, BinaryOperation.Operator.LBOUND)
+    assert ("Expecting integer datatype but found 'DataType.REAL'"
+            in str(excinfo.value))
+
+    operator = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND,
+        Reference(symbol), Literal("2", DataType.INTEGER))
+    my_range = Range.create(operator, one)
+    array_reference = Array.create(symbol, [my_range])
+    with pytest.raises(InternalError) as excinfo:
+        check_bound(array_reference, 1, 0, BinaryOperation.Operator.LBOUND)
+    assert ("Expecting literal value '2' to be the same as the current "
+            "array dimension '1'." in str(excinfo.value))
+
+    operator = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND,
+        Reference(symbol), Literal("1", DataType.INTEGER))
+    my_range = Range.create(operator, one)
+    array_reference = Array.create(symbol, [my_range])
+    check_bound(array_reference, 1, 0, BinaryOperation.Operator.LBOUND)
+
+
+# Function check_literal
```

Comment:
You don't need this comment (it's repeated in the docstring).

Reply:
Removed.